rotkehlchen/chain/ethereum/modules/liquity/trove.py

```diff
@@ -0,0 +1,506 @@
+                        debt_delta=debt_change,
+                        collateral_delta=collateral_change,
+                        trove_operation=operation,
+                    )
+                    result[owner]['trove'].append(event)
+                except KeyError as e:
+                    self.msg_aggregator.add_warning(
+                        f'Ignoring Liquity Trove event in Liquity. '
+                        f'Failed to decode remote information. {str(e)}.',
+                    )
+                    continue
+
+        for stake in staked['lqtyStakes']:
+            owner = to_checksum_address(stake['id'])
+            for change in stake['changes']:
+                try:
+                    timestamp = change['transaction']['timestamp']
+                    if timestamp < from_timestamp:
+                        continue
+                    if timestamp > to_timestamp:
+                        break
+                    operation_stake = LiquityStakeEventType.deserialize(change['stakeOperation'])
+                    stake_event = LiquityStakeEvent(
+                        kind='stake',
+                        tx=change['transaction']['id'],
+                        address=owner,
+                        timestamp=timestamp,
+                        stake_after=deserialize_optional_fval(change['stakedAmountAfter'], 'stakedAmountAfter', 'liquity'),  # noqa: E501
+                        stake_change=deserialize_optional_fval(change['stakedAmountChange'], 'stakedAmountChange', 'liquity'),  # noqa: E501
+                        issuance_gain=deserialize_optional_fval(change['issuanceGain'], 'issuanceGain', 'liquity'),  # noqa: E501
+                        redemption_gain=deserialize_optional_fval(change['redemptionGain'], 'redemptionGain', 'liquity'),  # noqa: E501
+                        stake_operation=operation_stake,
+                    )
+                    result[owner]['stake'].append(stake_event)
+                except KeyError as e:
+                    self.msg_aggregator.add_warning(
+                        f'Ignoring Liquity Trove event in Liquity. '
+                        f'Failed to decode remote information. {str(e)}.',
+                    )
+                    continue
+        return result
+
+    def get_history_events(
+        self,
+        from_timestamp: Timestamp,
+        to_timestamp: Timestamp,
+        addresses: List[ChecksumEthAddress],
+    ) -> List[DefiEvent]:
+        query = self._get_raw_history(addresses, 'trove')
+        result = []
+        for trove in query['troves']:
+            changes = self._process_trove_events(trove['changes'], from_timestamp, to_timestamp)
+            result.append(changes)
+        # Flatten the result (list of lists to list)
+        return reduce(add, result)
+
+    # -- Methods following the EthereumModule interface -- #
+    def on_startup(self) -> None:
+        pass
+
+    def on_account_addition(self, address: ChecksumEthAddress) -> None:
```

Comment:
Shouldn't you do a balance query for an account when it's added and this module is activated?